### PR TITLE
Graduate PSReadLine feature and add UseLegacyReadLine

### PR DESCRIPTION
**Pull request by TylerLeonhardt**
This stops checking for PSReadLine feature flag in the feature flags passed in.

Instead, it plumbs a `UseLegacyReadLine` parameter which allows the ability to force using the Legacy ReadLine experience (for situations like accessibility that require removing PSRL).

I have a separate PR on the vscode-powershell side taking advantage of this here: https://github.com/PowerShell/vscode-powershell/pull/2262

### CODE REVIEW

File: src/PowerShellEditorServices/Services/PowerShellContext/PowerShellContextService.cs [hunk 1]
```diff
@@ -171,19 +171,26 @@ public RunspaceDetails CurrentRunspace
...
             var logger = factory.CreateLogger<PowerShellContextService>();
 
-            // PSReadLine can only be used when -EnableConsoleRepl is specified otherwise
-            // issues arise when redirecting stdio.
+            // We should only use PSReadLine if we specificied that we want a console repl
+            // and we have not explicitly said to use the legacy ReadLine.
+            // We also want it if we are either:
+            // * On Windows on any version OR
+            // * On Linux or macOS on any version greater than or equal to 7
+            var shouldUsePSReadLine = enableConsoleRepl && !useLegacyReadLine
```
Comment (rjmholt) [inline src/PowerShellEditorServices/Services/PowerShellContext/PowerShellContextService.cs]:
```suggestion
            bool shouldUsePSReadLine = enableConsoleRepl && !useLegacyReadLine
```

File: src/PowerShellEditorServices/Services/PowerShellContext/PowerShellContextService.cs [hunk 2]
```diff
@@ -171,19 +171,26 @@ public RunspaceDetails CurrentRunspace
...
+                && (VersionUtils.IsWindows || (!VersionUtils.IsWindows && !VersionUtils.IsPS6));
```
Comment (rjmholt) [inline src/PowerShellEditorServices/Services/PowerShellContext/PowerShellContextService.cs]: Shouldn't `(IsWindows || (!IsWindows && IsPS6)` be the same as `(IsWindows || !IsPS6)`?

File: src/PowerShellEditorServices/Server/StdioPsesLanguageServer.cs [hunk 3]
```diff
@@ -23,7 +23,9 @@ internal class StdioPsesLanguageServer : PsesLanguageServer
             ProfilePaths profilePaths) : base(
                 factory,
                 minimumLogLevel,
-                // Stdio server can't support an integrated console so we pass in false.
+                // Stdio server can't support an integrated console so we pass in false for
+                // enableConsoleRepl and useLegacyReadLine.
+                false,
```
Comment (rjmholt) [inline src/PowerShellEditorServices/Server/StdioPsesLanguageServer.cs]: Possible to use named args here?

**Comment by rjmholt** [comment]
Hopefully we can slightly reduce the number of files a change like this touches in the binary cmdlet refactor!

File: src/PowerShellEditorServices/Services/PowerShellContext/PowerShellContextService.cs (hunk 2 above)
Comment (TylerLeonhardt) [reply src/PowerShellEditorServices/Services/PowerShellContext/PowerShellContextService.cs]: Good call - fixed!

File: src/PowerShellEditorServices/Server/StdioPsesLanguageServer.cs (hunk 3 above)
Comment (TylerLeonhardt) [reply src/PowerShellEditorServices/Server/StdioPsesLanguageServer.cs]: Fixed!

File: src/PowerShellEditorServices/Services/PowerShellContext/PowerShellContextService.cs (hunk 1 above)
Comment (TylerLeonhardt) [reply src/PowerShellEditorServices/Services/PowerShellContext/PowerShellContextService.cs]: fixed!

**Pull request merged**